### PR TITLE
Update input configuration.

### PR DESCRIPTION
**Pull request by josiah-tesfu**
Updated the input configuration. Configuration is initially set through the editor. Once the node is deployed, configuration can be set dynamically from input. If any parameters from the input message are empty, the previously set configurations are used.

### CODE REVIEW

**Review by sjpark608 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -5,29 +5,43 @@
...
+- When configured to read cold junction temperature:
+  - **payload** _number_: Temperature reading of the cold junction sensor.
+- When configured to read both cold junction and linearized temperature:
+  - **payload** _array[coldJunctionTemp, linearizedTemp]_: An array containing both the cold junction and linearized temperature.
+
+Example output:
+```
+msg {
+  "payload": [24.1875, 23.0703125]
+}
```
Comment (farzadpanahi) [inline README.md:46]: let's simplify this. I don't think we need to pass this kind of information on what to read. @sjpark608 does it make sense to have any inputs for TC?

**Review by farzadpanahi — CHANGES_REQUESTED** [review]

**Review by naomiven — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (sjpark608) [reply README.md:46]: I think it doesn't make sense to choose which temperature to read. The end user should be only interested in reading the linearized temperature

File: edgepi-thermocouple.js [hunk 2]
```diff
@@ -1,43 +1,52 @@
...
-        // Init new tc
-        const tc = new rpc.TcService(transport)
-        if (tc){
-            console.debug("Thermocouple node initialized on: ", transport);
-            node.status({fill:"green", shape:"ring", text:"tc initialized"});
+    initializeNode(config).then((tc) => {
+      this.on("input", async function (msg, send, done) {
+        node.status({ fill: "green", shape: "dot", text: "input recieved" });
+        try {
+          output = msg.payload ?? output;
```
Comment (jimmy121192) [inline edgepi-thermocouple.js:13]: Question : what is the possible value of `msg.payload`?
In the Nullish Operator (??) here: This operator returns the right value (`output`) ONLY if the left value is either `null` or `undefined`.
So if the `msg.payload` is neither `null` nor `undefined` (empty string , 0 ) this operator won't return the right value (`output`).
I would recommend to use `||` in this case

**Review by jimmy121192 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (josiah-tesfu) [reply README.md:46]: In that case, should I just ignore this pr? Besides adding input options, this pr doesn't include anything significant.

Comment (sjpark608) [reply README.md:46]: In that case yes we should ignore the PR

Comment (jimmy121192) [reply README.md]: Reverted

**Review by jimmy121192 — DISMISSED** [review]

File: README.md [hunk 3]
```diff
@@ -5,29 +5,28 @@
...
-  - **payload** *number*: Temperature reading of the cold junction sensor.
-  
+  - **payload** _number_: Temperature reading of the cold junction sensor.
 - When configured to read both cold junction and linearized temperature:
-  - **payload** *array[coldJunctionTemp, linearizedTemp]*: An array containing both the cold junction and linearized temperature.
-
-
-**NOTE:** Currently, EdgePi nodes are only available on x86 systems.
-
+  - **payload** _array[coldJunctionTemp, linearizedTemp]_: An array containing both the cold junction and linearized temperature.
```
Comment (farzadpanahi) [inline README.md:32]: let's fix this output readme as well. I think it should just say:
```
### Outputs
  - **payload** _number_:  TC temperature reading.
```

File: edgepi-thermocouple.html [hunk 4]
```diff
@@ -1,132 +1,159 @@
...
+      placeholder="Port"
+    />
+  </div>
+  <div class="form-row">
+    <label for="node-input-output">Output (&deg;C):</label>
+    <select id="node-input-output">
+      <option value="lin">Linearized Temperature</option>
+      <option value="cj">Cold-Junction Temperature</option>
+      <option value="cj-lin">Cold-Junction & Linearized Temperature</option>
+    </select>
```
Comment (farzadpanahi) [inline edgepi-thermocouple.html:106]: let's remove these selections. the user does not need to select this.

File: edgepi-thermocouple.html [hunk 5]
```diff
@@ -1,132 +1,159 @@
...
+  </p>
+  <dl class="message-properties">
+    <dt>
+      payload
+      <span class="property-type">array[coldJunctionTemp, linearizedTemp]</span>
+    </dt>
+    <dd>
+      an array containing both the cold junction and linearized temperature.
+    </dd>
+  </dl>
```
Comment (farzadpanahi) [inline edgepi-thermocouple.html:149]: update the output section similar to what we have in readme

File: edgepi-thermocouple.js [hunk 6]
```diff
@@ -1,43 +1,52 @@
...
+      this.on("input", async function (msg, send, done) {
+        node.status({ fill: "green", shape: "dot", text: "input recieved" });
+        try {
+          output = msg.payload || output;
+          let temps = await tc.singleSample();
+          if (output === "cj-lin") {
+            msg.payload = temps;
+          } else {
+            msg.payload = output === "cj" ? temps[0] : temps[1];
+          }
```
Comment (farzadpanahi) [inline edgepi-thermocouple.js:40]: we should only write  linearized temp to payload.

**Review by farzadpanahi — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (jimmy121192) [reply README.md]: Done

File: edgepi-thermocouple.html (hunk 4 above)
Comment (jimmy121192) [reply edgepi-thermocouple.html]: Done

File: edgepi-thermocouple.html (hunk 5 above)
Comment (jimmy121192) [reply edgepi-thermocouple.html]: Done

File: edgepi-thermocouple.js (hunk 6 above)
Comment (jimmy121192) [reply edgepi-thermocouple.js]: DOne

File: edgepi-thermocouple.js [hunk 7]
```diff
@@ -33,10 +33,13 @@ module.exports = function (RED) {
...
           output = msg.payload || output;
           let temps = await tc.singleSample();
-          if (output === "cj-lin") {
+          // if (output === "cj-lin") {
+          //   msg.payload = temps;
+          // } else {
+          //   msg.payload = output === "cj" ? temps[0] : temps[1];
+          // }
+          if (output === "lin") {
             msg.payload = temps;
```
Comment (farzadpanahi) [inline edgepi-thermocouple.js:42]: I think the payload should be `temps[1]`. check the sdk or test it.

**Review by farzadpanahi — CHANGES_REQUESTED** [review]

File: edgepi-thermocouple.js (hunk 7 above)
Comment (jimmy121192) [reply edgepi-thermocouple.js]: updated

File: edgepi-thermocouple.js [hunk 8]
```diff
@@ -1,55 +1,47 @@
...
           let temps = await tc.singleSample();
-          // if (output === "cj-lin") {
-          //   msg.payload = temps;
-          // } else {
-          //   msg.payload = output === "cj" ? temps[0] : temps[1];
-          // }
-          if (output === "lin") {
-            msg.payload = temps;
-          }
+          msg.payload = temps[0];
```
Comment (farzadpanahi) [inline edgepi-thermocouple.js:15]: are you sure linearized temp is at `temps[0]` ?

**Review by farzadpanahi — CHANGES_REQUESTED** [review]

File: edgepi-thermocouple.js (hunk 8 above)
Comment (jimmy121192) [reply edgepi-thermocouple.js]: oh wait. it should be temps[1]. Fixed

**Review by farzadpanahi — APPROVED** [review]

**Comment by jimmy121192** [comment]
I have checked the source code on edgepi-sdk, and it is the 2nd value
Screen Shot 2024-01-12 at 2 06 44 PM

**Pull request merged**